src/api/procedures/registerIdentity.ts

```diff
@@ -0,0 +1,48 @@
+import { Identity } from '~/api/entities';
+import { Procedure } from '~/base';
+import { Role, RoleType, SigningKey } from '~/types';
+import { dateToMoment, signingKeyToMeshSigningKey, valueToDid } from '~/utils';
+
+export interface RegisterIdentityParams {
+  target: string | Identity;
+  expiry?: Date;
+  signingKeys?: SigningKey[];
+}
+
+/**
+ * @hidden
+ */
+export async function prepareRegisterIdentity(
```

Comment:
This procedure should return the newly created Identity. You can use a resolver to catch the event (`DidCreated`) and get the new DID